### PR TITLE
Btm 585 tear down script fix

### PR DESCRIPTION
**Pull request by NKannan14**
created separate function for listingStack resources
replaced DescribeStackResourcesCommand with ListStackResourcesCommand for pagination as in the AWS docs it suggested to ListStackResourcesCommand instead of DescribeStackResourcesCommand for pagination https://awscli.amazonaws.com/v2/documentation/api/latest/reference/cloudformation/describe-stack-resources.html
update type StackResource to StackResourceSummary as StackRescource not exists in ListStackResourcesCommand
Added condition to do tryToDelete

### CODE REVIEW

**Review by markschnitziusgovuk — APPROVED** [review]
LGTM

File: src/tools/teardown-cf-stack.ts [hunk 1]
```diff
@@ -149,20 +149,44 @@ const destroyStack = async (stackName: string): Promise<StackResource[]> => {
...
+): Promise<StackResourceSummary[]> => {
+  const deleteResult = await cfClient.send(
+    new DeleteStackCommand({ StackName: stackName })
+  );
+  if (isAWSError(deleteResult)) {
+    console.log(deleteResult.error.toString());
+    process.exit(-1);
+  }
+  let resourcesFailed: StackResourceSummary[] = [];
+  while (true) {
```
Comment (Si3rr4wow) [inline src/tools/teardown-cf-stack.ts:174]: For anyone else who gets spooked seeing this here (or even to myself if I circle back to this) this look is broken by invoking `process.exit`

File: src/tools/teardown-cf-stack.ts [hunk 2]
```diff
@@ -149,20 +149,44 @@ const destroyStack = async (stackName: string): Promise<StackResource[]> => {
...
+  stackName: string
+): Promise<StackResourceSummary[]> => {
+  const deleteResult = await cfClient.send(
+    new DeleteStackCommand({ StackName: stackName })
+  );
+  if (isAWSError(deleteResult)) {
+    console.log(deleteResult.error.toString());
+    process.exit(-1);
+  }
+  let resourcesFailed: StackResourceSummary[] = [];
```
Comment (Si3rr4wow) [inline src/tools/teardown-cf-stack.ts:173]: I think you can get rid of this declaration and keep the old `const resourcesFailed = result.StackResources.filter(` one. It doesn't seem like you need access to this `resourcesFailed` outside the `while` loop

Comment (NKannan14) [reply src/tools/teardown-cf-stack.ts]: Yes, thanks Mark. Changed to local variable

**Pull request merged**